tests/utils.py

```diff
@@ -24,3 +25,11 @@ def __init__(self, mode: str = MetricAggregationMode.per_segment, alpha: float =
     @property
     def name(self) -> str:
         return self.__repr__()
+
+
+def equals_with_nans(first_df: pd.DataFrame, second_df: pd.DataFrame) -> bool:
```

Comment:
you can use smth like [np.allclose](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html)

Reply:
It isn't working with NaNs (result is False), it is also hard to use if dataframes contains str types like segments.